core/src/actors/blocks_manager/mod.rs

```diff
@@ -40,6 +63,11 @@ pub mod messages;
 pub struct BlocksManager {
     /// Blockchain information data structure
     chain_info: Option<ChainInfo>,
+    /// Map that relates an epoch with the hashes of the blocks for that epoch
+    // One epoch can have more than one block
+    epoch_to_block_hash: HashMap<Epoch, HashSet<Hash>>,
```

Comment:
Just for the record: we will have to change this in future sprints so as to clarify:
- During epoch _E<sub>t</sub>_, multiple block candidates _B<sub>t<sub>i</sub></sub>_ may be produced and distributed by different parties.
- Every block candidate _B<sub>t<sub>i</sub></sub>_ contains:
  - A checkpoint beacon linking checkpoint _t_ with the hash of the block for the previous epoch _B<sub>t-1</sub>_.
  - A valid proof of eligibility for checkpoint _t_.
  - A list of transactions produced and broadcast before checkpoint _t_ (epochs _E<sub>0...t</sub>_).
- Block candidates need to be accepted, validated and kept in memory for the whole duration of _E<sub>t</sub>_.
- Once epoch _E<sub>t</sub>_ is over (checkpoint _t+1_):
  - No more block candidates will be accepted for _E<sub>t</sub>_ 
  - All block candidates for _E<sub>t</sub>_ need to be merged together using a function that will produce a single consolidated block _B<sub>t</sub>_ in a deterministic manner (YTBD).
- Consolidated blocks can immediately be appended to the block chain and persisted into storage.
- The hash of the consolidated block _B<sub>t</sub>_ is the one that block candidates for epoch _E<sub>t+1</sub>_ will use in their checkpoint beacons along with the number of checkpoint _t+1_.